app/src/androidTest/java/com/github/gogetters/letsgo/activities/ProfileActivityTest.kt

```diff
@@ -0,0 +1,253 @@
+        val device = UiDevice.getInstance(InstrumentationRegistry.getInstrumentation())
+        var waitButton = device.findObject(UiSelector().textContains("wait"))
+        if (waitButton.exists()) {
+            waitButton.click()
+        }
+    }
+
+    @After
+    fun cleanUp() {
+        Intents.release()
+        scenario.close()
+    }
+
+    @Test
+    fun dialogOpensOnProfileClick() {
+        scenario = ActivityScenario.launch(intent)
+        sleep()
+        onView(withId(R.id.profile_imageView_image)).perform(click())
+        onView(withText(R.string.profile_dialogTitle)).check(matches(isDisplayed()))
+    }
+
+    @Test
+    fun cameraIntentIsFired() {
+        scenario = ActivityScenario.launch(intent)
+        sleep()
+        onView(withId(R.id.profile_imageView_image)).perform(click())
+        clickAtIndex(0, "Take Picture")
+        acceptPermissions()
+        Intents.intended(IntentMatchers.hasAction(MediaStore.ACTION_IMAGE_CAPTURE))
+    }
+
+    @Test
+    fun galleryIntentIsFired() {
+        scenario = ActivityScenario.launch(intent)
+        sleep()
+        onView(withId(R.id.profile_imageView_image)).perform(click())
+
+        //onView(withText(R.string.profile_chooseFromGallery)).perform(click())
+        clickAtIndex(1, "Choose from Gallery")
+        acceptPermissions()
+        Intents.intended(IntentMatchers.hasAction(Intent.ACTION_CHOOSER))
+    }
+
+    @Test
+    fun profilePictureDialogDisappearsOnCancel() {
+        scenario = ActivityScenario.launch(intent)
+        sleep()
+        onView(withId(R.id.profile_imageView_image)).perform(click())
+        //onView(withText(R.string.profile_cancel)).perform(click())
+        clickAtIndex(2, "Cancel")
+    }
+
+    //@Test doesn't work
+    fun gallerySelectionIsWellPlaced() {
+        scenario = ActivityScenario.launch(intent)
+        scenario.onActivity {
+            savePickedImage(it)
+            val imgGalleryResult = createImageGallerySetResultStub(it)
+            intending(hasAction(Intent.ACTION_CHOOSER)).respondWith(imgGalleryResult)
+            //auctionPhotos_CreationInitialUI()
+            onView(withId(R.id.profile_imageView_image)).perform(click())
+            clickAtIndex(1, "Choose from Gallery")
+            acceptPermissions()
+            onView(withId(R.id.profile_imageView_image)).check(matches(hasImageSet()))
+        }
+    }
+
+    //@Test doesn't work
+    fun capturedImageIsWellPlaced() {
+        scenario = ActivityScenario.launch(intent)
+        scenario.onActivity {
+            val imgCaptureResult = createImageCaptureActivityResultStub(it)
+            intending(hasAction(MediaStore.ACTION_IMAGE_CAPTURE)).respondWith(imgCaptureResult)
+
+            onView(withId(R.id.profile_imageView_image)).perform(click())
+            clickAtIndex(0, "Take Picture")
+            acceptPermissions()
+            onView(withId(R.id.profile_imageView_image)).check(matches(hasImageSet()))
+        }
+    }
+
+    // @Test
```

Comment:
I noticed code coverage is a bit too low. We could now use the mock database to run this test and bring coverage up!